packages/contracts/src/contracts/current/protocol/Exchange/IExchange.sol

```diff
@@ -35,8 +35,8 @@ contract IExchange {
         address indexed maker,
         address taker,
         address indexed feeRecipient,
-        address makerToken,
-        address takerToken,
+        bytes makerAssetData,
```

Comment:
This converts to `keccak256(assetProxyId, tokenAddress)` right? Which adds little bit of complexity to watch for Token pair trades (especially if there is a mix of old proxy and new proxy). There are multiple combinations of ZRX/WETH to watch for (due to multiple asset proxies). In the case of ERC721 this is `keccak256(assetProxyId, tokenAddress, tokenId)`, which makes it difficult to watch all ERC721 exchanges (since one needs the tokenId).

The combinations also increase if we throw in the fee ZRX_PROXY_DATA in there as well. 

This is not insurmountable at all, just something we would have to document well and provide guides on. What is the gas cost/tradeoff of keeping both here?

Reply:
Can confirm that the `makerAssetData`/`takerAssetData` fields are logged in place of `makerToken` and `takerToken`.

@abandeali1 I'm open to moving the tokens back out of the byte arrays. I think I'm just missing the full scope of this approach. From my understanding, it seems we can better generalize the exchange by abstracting away proxy-specific data. When we start exchanging other assets, like ENS names, fields like makerToken and takerToken don't appear to provide value at the exchange level.

Reply:
I must've gotten confuse din the solidity docs, maybe the hashing only applies to indexed data?

This event is more for the general ecosystem to read. Is the concern the fact we don't parse this unstructured byte array in the Exchange contract and that this would cause unnecessary struct knowledge just to log an event with particular fields?